includes/admin/class-wc-admin-notices.php

```diff
@@ -530,6 +548,61 @@ public static function simplify_commerce_notice() {
 	public static function theme_check_notice() {
 		wc_deprecated_function( 'WC_Admin_Notices::theme_check_notice', '3.3.0' );
 	}
+
+	/**
+	 * Check if uploads directory is protected.
+	 *
+	 * @since 4.2.0
+	 * @return bool
+	 */
+	protected static function is_uploads_directory_protected() {
+		$cache_key = '_woocommerce_upload_directory_status';
+		$status    = get_transient( $cache_key );
+
+		// Check for cache.
+		if ( false !== $status ) {
+			return 'protected' === $status;
+		}
+
+		// Get uploads directory data and allows to get created if doesn't exists.
+		$uploads = wp_upload_dir( null, true );
+
+		// Skip if returns an error.
+		if ( $uploads['error'] ) {
+			return false;
+		}
+
+		// Allow us to easily interact with the filesystem.
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		WP_Filesystem();
+		global $wp_filesystem;
+
+		$is_protected  = false;
+		$test_dir      = 'woocommerce-uploads-test';
+		$test_dir_path = trailingslashit( $uploads['basedir'] ) . $test_dir;
+
+		// Clean up the test directory before we start.
+		if ( $wp_filesystem->exists( $test_dir_path ) ) {
+			$wp_filesystem->delete( $test_dir_path );
+		}
+
+		// Create a new directory to check in case the uploads root is only protected by an index.php or index.html file.
+		if ( $wp_filesystem->mkdir( $test_dir_path ) ) {
```

Comment:
If file permissions don't allow the creation of the test directory at this location, then `$is_protected` will be `false`, so it'll throw the warning, even though it's not necessarily unprotected.